test/test_oneagentmetadataenricher.py

```diff
@@ -143,7 +136,16 @@ def test_indirection_file_empty(self):
             self.assertEqual(0, len(res))
 
     def test_get_file_contents_success(self):
-        mock = mock_open(read_data="key1=value1\nkey2=value2")
+        # this only tests _get_metadata_file_content() paritally since both
+        # open calls are mocked, so even if the file does not exist this test
+        # will pass since the second open call is also mocked.
+        mock = mock_open()
+        # return "indirection_filename" on the first open call and the
+        # key-vlaue pairs on the second.
+        mock.side_effect = [mock_open(read_data=x).return_value for x in
```

Comment:
Is the side effect evaluated before or after it returns `read_data`?

Reply:
the object named `mock` does not have a `read_data` property in this case. therefore, on the first `open` call, `mock_open(read_data="indirection_filename").return_value` is returned and on the second `open` call `mock_open(read_data="key1=value1\nkey2=value2").return_value` is returned, so the object called `mock` is like a wrapper for the other two.